### ISSUE TITLE
[Feature Request]Hope SB3 can support DQfD

### ISSUE THREAD

**Issue by nianba** [DLR-RM/stable-baselines3#1506]
### 🚀 Feature

Support for Deep Q-learning from Demonstrations algrithms.

### Motivation

Deep Q-learning from Demonstrations (DQfD) has good performance when you have small amounts of demonstration data.
Will you consider adding DQFD to the algorithm library?
Thanks！

### Alternatives
Modifying the training function and replay buffers may works.
[paper](https://arxiv.org/pdf/1704.03732.pdf)
a implementation of dqfd：https://github.com/go2sea/DQfD

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo

**Comment by araffin** [comment]
Hello,
Please fill out the template completely (notably the "alternatives" section).

I'm not sure to recall, DQNfD has two replay buffers? or an additional loss?

**Comment by nianba** [comment]
Thank you for the reminder. I have filled the alternatives.
Compared to DQN, DQFD has the following improvements:
1. During the pre-training phase, only expert demonstration data is used for supervised learning. Four types of losses are used to optimize the neural network, with the goal of producing a decent initial policy.
2. During the online learning phase, Prioritized Experience Replay (PER) is used to sample from a mixture of expert data and self-generated data obtained through real-time interactions to further improve the policy.

**Comment by nianba** [comment]
> 

any progress？

**Comment by araffin** [comment]
Hello,
sorry for the delayed answer.
After thinking more about that, I would say DQfD should be a better candidate for https://github.com/HumanCompatibleAI/imitation or for a separate repo.

**Comment by nianba** [comment]
thanks for the advice！
